File: Models/DataBase/StockTransaction.cs
using System.ComponentModel.DataAnnotations;

namespace quan_ly_chuoi_cua_hang_tap_hoa_api.Models.DataBase
{
    public class StockTransaction
    {
        [Key]
        public int TransactionId { get; set; }
        public int WarehouseId { get; set; }
        public int ProductId { get; set; }
        public int Quantity { get; set; }
        public string TransactionType { get; set; }
        public int? ReferenceId { get; set; }
        public string CreatedBy { get; set; }
        public DateTime CreatedAt { get; set; }

        public virtual Warehouse Warehouse { get; set; }
        public virtual Product Product { get; set; }
        public virtual Users User { get; set; }
    }
}

File: lib/models/stock_transaction.dart
/// Model giao dịch kho
class StockTransaction {
  final int transactionId;
  final int warehouseId;
  final int productId;
  final String productName;
  final String transactionType;
  final int quantity;
  final String createdBy;
  final DateTime createdAt;

  StockTransaction({
    required this.transactionId,
    required this.warehouseId,
    required this.productId,
    required this.productName,
    required this.transactionType,
    required this.quantity,
    required this.createdBy,
    required this.createdAt,
  });

  factory StockTransaction.fromJson(Map<String, dynamic> json) =>
      StockTransaction(
        transactionId: json['transactionId'] ?? 0,
        warehouseId: json['warehouseId'] ?? 0,
        productId: json['productId'] ?? 0,
        productName: json['productName'] ?? '',
        transactionType: json['transactionType'] ?? '',
        quantity: json['quantity'] ?? 0,
        createdBy: json['createdBy'] ?? '',
        createdAt: DateTime.tryParse(json['createdAt'] ?? '') ?? DateTime.now(),
      );

  Map<String, dynamic> toJson() => {
    'transactionId': transactionId,
    'warehouseId': warehouseId,
    'productId': productId,
    'productName': productName,
    'transactionType': transactionType,
    'quantity': quantity,
    'createdBy': createdBy,
    'createdAt': createdAt.toIso8601String(),
  };
}
